### PR TITLE
better segment navigation

### PR DESCRIPTION
**Pull request by yhedholm**
This change sorts unfinished segments ahead of all finished segment. (An unfinished segment is one where no children have been specified.) Unfinished segments are also marked with a leading '*'.

### CODE REVIEW

File: frontend/rolecall/src/app/piece/piece_editor.component.ts [hunk 1]
```diff
@@ -36,7 +45,8 @@ type WorkingPiece = Piece & {
 export class PieceEditor implements OnInit {
   dragAndDropData: DraggablePosition[] = [];
   currentSelectedPiece: WorkingPiece;
-  renderingPieces: WorkingPiece[];
+  workingPieces: WorkingPiece[];
+  renderingItems: RenderingItem[];
```
Comment (mikzat) [inline frontend/rolecall/src/app/piece/piece_editor.component.ts:49]: Add comments for `workingPieces` and `renderingItems`, from variable names it's not obvious what they are for.

File: frontend/rolecall/src/app/piece/piece_editor.component.ts [hunk 2]
```diff
@@ -75,14 +85,31 @@ export class PieceEditor implements OnInit {
     this.pieceAPI.getAllPieces();
   }
 
+  private buildRenderingList() {
+    this.renderingItems = this.workingPieces.map((wpiece, wpieceIx) => {
```
Comment (mikzat) [inline frontend/rolecall/src/app/piece/piece_editor.component.ts:89]: `(piece, pieceIndex) => {`

File: frontend/rolecall/src/app/piece/piece_editor.component.ts [hunk 3]
```diff
@@ -75,14 +85,31 @@ export class PieceEditor implements OnInit {
...
+      const missingChidren = wpiece.type === 'SEGMENT' ? false : wpiece.positions.length === 0;
```
Comment (mikzat) [inline frontend/rolecall/src/app/piece/piece_editor.component.ts:90]: `missingChildren` sounds like I would expect it to contain a list of "child" items, not a boolean.

How about something like `hasNoChildren`

Or even better, invert the negation to reduce complexity and make it
`const hasChildren = ...`

File: frontend/rolecall/src/app/piece/piece_editor.component.ts [hunk 4]
```diff
@@ -75,14 +85,31 @@ export class PieceEditor implements OnInit {
...
+      const prefix = missingChidren ? ' ' : 'z';
```
Comment (mikzat) [inline frontend/rolecall/src/app/piece/piece_editor.component.ts:91]: `sortPrefix`, because it's not actually a prefix for the name

**Review by mikzat — CHANGES_REQUESTED** [review]

File: frontend/rolecall/src/app/piece/piece_editor.component.ts (hunk 4 above)
Comment (yhedholm) [reply frontend/rolecall/src/app/piece/piece_editor.component.ts:91]: Done.

File: frontend/rolecall/src/app/piece/piece_editor.component.ts (hunk 3 above)
Comment (yhedholm) [reply frontend/rolecall/src/app/piece/piece_editor.component.ts:90]: Done. I prefer hasNoChildren to hasChildren. We are looking for the pieces/segments that have no children.

File: frontend/rolecall/src/app/piece/piece_editor.component.ts (hunk 2 above)
Comment (yhedholm) [reply frontend/rolecall/src/app/piece/piece_editor.component.ts:89]: Done. I prefer workingPiece to piece. Piece is a standard and ubiquitous data structure and WorkingPiece is a bigger structure than contains transient fields supporting the editing process.

File: frontend/rolecall/src/app/piece/piece_editor.component.ts (hunk 1 above)
Comment (yhedholm) [reply frontend/rolecall/src/app/piece/piece_editor.component.ts:49]: Done.

File: frontend/rolecall/src/app/piece/piece_editor.component.ts [hunk 5]
```diff
@@ -86,17 +89,19 @@ export class PieceEditor implements OnInit {
   }
 
   private buildRenderingList() {
-    this.renderingItems = this.workingPieces.map((wpiece, wpieceIx) => {
-      const missingChidren = wpiece.type === 'SEGMENT' ? false : wpiece.positions.length === 0;
-      const prefix = missingChidren ? ' ' : 'z';
-      const name = missingChidren ? '*' + wpiece.name : wpiece.name;
+    this.renderingItems = this.workingPieces.map(
+        (workinPiece, workingPieceIndex) => {
```
Comment (mikzat) [inline frontend/rolecall/src/app/piece/piece_editor.component.ts:93]: spelling, `workinPiece` -> `workingPiece`

**Review by mikzat — APPROVED** [review]

**Pull request merged**